graylog2-server/src/main/java/org/graylog/plugins/views/storage/migration/state/machine/MigrationStateMachineBuilder.java

```diff
@@ -73,6 +73,7 @@ private static StateMachineConfig<MigrationState, MigrationStep> configureStates
         config.configure(MigrationState.PROVISION_DATANODE_CERTIFICATES_PAGE)
                 .permit(MigrationStep.PROVISION_DATANODE_CERTIFICATES, MigrationState.PROVISION_DATANODE_CERTIFICATES_RUNNING, migrationActions::provisionAndStartDataNodes);
 
+        // This page should contain the "Please restart Graylog to continue with data migration"
```

Comment:
Remote Reindexing isn't going to work this way, I am afraid. 
After provisioning is finished, the connection string needs to be removed and Graylog needs to be restarted to connect to the data nodes.

Reply:
this should work now after the change i pushed. we just have to make the user aware that he needs to remove the connection string when we tell him to restart